distributed/bokeh/scheduler.py

```diff
@@ -109,13 +109,15 @@ def __init__(self, scheduler):
     def update(self):
         with log_errors():
             s = self.scheduler
-            d = {'Tasks': [len(s.tasks)],
-                 'Stored': [len(s.who_has)],
-                 'Processing': ['%d / %d' % (len(s.rprocessing), s.total_ncores)],
-                 'Waiting': [len(s.waiting)],
-                 'No Worker': [len(s.unrunnable)],
-                 'Erred': [len(s.exceptions)],
-                 'Released': [len(s.released)]}
+            freqs = frequencies(ts.state for ts in s.task_states.values())
```

Comment:
This could get ugly when we get to a million tasks (which we should treat as normal)

```python
In [1]: class A(object):
   ...:     pass
   ...: 

In [2]: objs = []

In [3]: for i in range(1000000):
   ...:     obj = A()
   ...:     obj.x = 'a'
   ...:     objs.append(obj)
   ...:     

In [4]: from toolz import frequencies

In [5]: %time _ = frequencies([o.x for o in objs])
CPU times: user 140 ms, sys: 0 ns, total: 140 ms
Wall time: 139 ms
```

So we might have to drop this diagnostic

Reply:
Some of these numbers we can get from other state, right? 

    {'memory': scheduler.who_has, 'erred': scheduler.exceptions_blame, ...}

Reply:
That other state are just pseudo-mappings that query the internal `task_states`.
If the diagnostic is important, though, we can always maintain aggregates in the scheduler. It shouldn't be very costly (but, of course, always risks losing sync if there's a bug :-)).

Reply:
See https://github.com/dask/distributed/pull/1594/files#diff-6a3108098afcddf88b38652ae1e383b5R240 for what I mean.

Reply:
What do you think: should we keep running counts in the scheduler or should we remove those diagnostics?

Reply:
I think that it depends on how many cases there are like this.  If this is the only one then lets just remove the diagnostic

Reply:
In worst-case we can move this out to a counting plugin.

Reply:
There's `CountsJSON` and `StateTable`.

Reply:
I recommend that we stop using `StateTable` and leave `CountsJSON` as you have it now.  `StateTable` is called periodically (500ms), `CountsJSON` is up to the user. 

cc @bluenote10 I believe that this affects you.  Note that this operation will start becoming more expensive.  If you have million+ node graphs then I recommend not calling this route more often than once a second.